rskj-core/src/main/java/org/ethereum/vm/PrecompiledContracts.java

```diff
@@ -215,7 +221,22 @@ public abstract static class PrecompiledContract {
 
         public abstract long getGasForData(byte[] data);
 
+        @Deprecated
         public void init(Transaction tx, Block executionBlock, Repository repository, BlockStore blockStore, ReceiptStore receiptStore, List<LogInfo> logs) {
```

Comment:
```suggestion
        public final void init(Transaction tx, Block executionBlock, Repository repository, BlockStore blockStore, ReceiptStore receiptStore, List<LogInfo> logs) {
```
- I think we should mark it as `final` just to prohibit child classes to override this method
- we should probably call the newer version of `init()` in the `TransactionExecutor` and `Program` classes and provide an instance of `ProgramInvoke` in there?